File: 002_AgentesDeResolucaoDeProblemas/src/No.java
public class No {
    public Estado estado;
    public No pai;
    public int custoCaminho;

    public No(Estado estado, No pai, int custoCaminho) {
        this.estado = estado;
        this.pai = pai;
        this.custoCaminho = custoCaminho;
    }

    public No(No no) {
        this.estado = no.estado;
        this.pai = no.pai;
        this.custoCaminho = no.custoCaminho;
    }
}

File: 002_AgentesDeResolucaoDeProblemas/No.py
class No:
    def __init__(self, estado=None, pai=None, custo=None):
        self.estado = estado
        self.pai = pai
        self.custo = custo

    def __str__(self):
        return f"Estado: {self.estado.nome} \nCusto: {self.custo} \nPai: {self.pai.estado.nome if self.pai else 'None'}"